applications/segment_everything/README.md

```diff
@@ -0,0 +1,134 @@
+#### tensorRT version
+To use the trt engine files in holoscan later, we need to have the matching tensorrt version. 
+In this case we are looking for the version that matches 
+- os="ubuntu2004"
+- tag="8.6.1-cuda-12.0"
+- arm based system (using a CAGX devkit)
+
+following the instructions on https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-861/install-guide/index.html#downloading for the debian installation procedure.
+
+The tensorRT version needs to be downloaded from https://developer.nvidia.com/nvidia-tensorrt-8x-download
+
+In this case I download it to a folder on the m2 drive. e.g. 
+```sh
+mkdir /media/m2/software/tensorrt
+cd /media/m2/software/tensorrt
+
+wget https://developer.nvidia.com/downloads/compute/machine-learning/tensorrt/secure/8.6.1/local_repos/nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_arm64.deb
+```
+
+#### run docker container
+Now the pytorch container can be run. Mount <HOLOHUB_SOURCE_DIR>, which contains the holohub repo, and mount the software folder, which contains the downloaded trt .deb installation file.
+
+```sh
+docker run --rm -it -v $PWD:/workspace -v /media/m2/software:/workspace/software nvcr.io/nvidia/pytorch:23.04-py3 /bin/bash
+```
+##### installation of tensorrt 
+follow the commandline steps below to install tensorRT and additional packages.
+```sh
+cd /workspace/software/tensorrt
+dpkg -i nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0_1.0-1_arm64.deb
+
+cp /var/nv-tensorrt-local-repo-ubuntu2004-8.6.1-cuda-12.0/nv-tensorrt-local-7148CA18-keyring.gpg /usr/share/keyrings/
+
+apt-get update
+
+apt-get install tensorrt
+apt-get install python3-libnvinfer-dev
+python3 -m pip install numpy onnx onnx-graphsurgeon
+```
+
+Check the installed version 
+```sh
+dpkg-query -W tensorrt
+```
+
+## 2. create trt engine files
+
+In the same docker container we can now proceed with building the trt engine files.
+First, pull the sam_trt_light repo, which contains a script to create the engine files.
+```
+cd /workspace
+git clone https://github.com/maximilianofir/sam_trt_light.git
+```
+
+Install the package sam_trt_light, and some additional requirements.
+```
+cd /workspace/sam_trt_light
+pip install -e .
+pip install onnxruntime onnx_graphsurgeon colored polygraphy --upgrade
+```
+
+Download the models, e,g, sam_vit_b:
+```
+mkdir downloads
+cd downloads
+wget https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth
+```
+I downloaded them to a downloads folder, e.g. "/workspace/sam_trt_light/downloads"
+
+Create folders for the onnx and engine files 
+
+```
+cd ..
+mkdir onnx engine
+```
+
+Run the trt_inference.py script and save the onnx and engine files to the created folders
+
+```
+python scripts/trt_inference.py --checkpoint=/workspace/sam_trt_light/downloads/sam_vit_b_01ec64.pth --input-image=images/apples.jpg --mode point --visualize --output-image=output.png --model-type vit_b --onnx-dir /workspace/sam_trt_light/onnx --engine-dir /workspace/sam_trt_light/engine
```

Comment:
Do we need an imput-image and visualization for the conversion?

Reply:
I'll try to avoid this additional repo, will ping you when I am ready.